sdk/identity/azure-identity/azure/identity/_internal/msal_credentials.py

```diff
@@ -86,3 +86,15 @@ def get_token(self, *scopes):
             raise ClientAuthenticationError(message="authentication failed: {}".format(result.get("error_description")))
 
         return AccessToken(result["access_token"], now + int(result["expires_in"]))
+
+
+class PublicClientCredential(MsalCredential):
+    """Wraps an MSAL PublicClientApplication with the TokenCredential API"""
+
+    def __init__(self, **kwargs):
+        # type: (Any) -> None
+        super(PublicClientCredential, self).__init__(
+            app_class=msal.PublicClientApplication,
+            authority="https://login.microsoftonline.com/" + kwargs.pop("tenant", "organizations"),
```

Comment:
Should the host of the authority be hard-coded? This changes depending on which cloud you are targeting, no?

Reply:
It should not. Sovereign cloud support is tracked by #5850. I'm waiting for some global configuration I can leverage to avoid implementing and later replacing something idiosyncratic.